cranelift-codegen/meta/src/cdsl/instructions.rs

```diff
@@ -533,6 +533,57 @@ impl Bindable for BoundInstruction {
     }
 }
 
+/// Checks that the input operands actually match the given format.
+fn verify_format(inst_name: &str, operands_in: &Vec<Operand>, format: &InstructionFormat) {
```

Comment:
Is this new code or does it replace something else?

Reply:
It is replacing code in FormatRegistry::lookup, that made sure to find the right format. Since we're now specifying it manually, a check is required to ensure it matches what the instruction actually needs.